File: src/harbor_clerk/api/schemas/source_ref.py
"""Shared source/citation response schemas."""

from typing import Literal

from pydantic import BaseModel


class SourceRefOut(BaseModel):
    doc_id: str
    doc_title: str
    chunk_id: str | None = None
    pages: str | None = None
    section: str | None = None
    source_kind: Literal["document", "email", "attachment", "unknown"]
    source_label: str
    folder_label: str | None = None
    relative_path: str | None = None
    citation: str

File: frontend/src/types/sourceRef.ts
export interface SourceRef {
  doc_id: string
  doc_title: string
  chunk_id?: string | null
  pages?: string | null
  section?: string | null
  source_kind: 'document' | 'email' | 'attachment' | 'unknown'
  source_label: string
  folder_label?: string | null
  relative_path?: string | null
  citation: string
}
